src/main/java/com/linkedin/kmf/services/SignalFxMetricsReporterService.java

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.linkedin.kmf.services;
+
+import static com.linkedin.kmf.common.Utils.getMBeanAttributeValues;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.linkedin.kmf.common.MbeanAttributeValue;
+import com.linkedin.kmf.services.configs.SignalFxMetricsReporterServiceConfig;
+import com.signalfx.codahale.metrics.SettableDoubleGauge;
+import com.signalfx.codahale.reporter.MetricMetadata;
+import com.signalfx.codahale.reporter.SignalFxReporter;
+import com.signalfx.endpoint.SignalFxEndpoint;
+
+public class SignalFxMetricsReporterService implements Service {
+  private static final Logger LOG = LoggerFactory.getLogger(SignalFxMetricsReporterService.class);
+
+  private final String _name;
```

Comment:
Is the "underscore for private variables" a convention in this project, or is this just yours?

Reply:
@mpetazzoni yes